lib/fabric/app/models/fabric/invoice.rb

```diff
@@ -9,6 +10,8 @@ class Invoice
       primary_key: :stripe_id, dependent: :destroy
     has_one :payment_intent, class_name: 'Fabric::PaymentIntent',
       primary_key: :stripe_id, dependent: :destroy
+    has_one :subscription, class_name: 'Fabric::Subscription',
+      primary_key: :latest_invoice_id
```

Comment:
Fabric::Subscription has a field called `latest_invoice_id` for the `latest_invoice` relationship. Typically the `primary_key` would be `invoice_id` on the subscription, but since it's under `latest_invoice`, I defined the `primary_key`. Does that make sense?

Reply:
No, it doesn't make sense.

A `has_one` relation can only have one primary key and one foreign key. The primary key and foreign key describe the whole relationship; they shouldn't be different on both sides. The primary key refers to the uniquely identifying value, and the foreign key is where it's stored. So in this case, you'd want the primary key to be the `Invoice`'s `stripe_id` and the foreign key to be the `latest_invoice_id` field on `Subscription`. The way you have it right now sets up different primary keys on each side of the relation.
See https://www.mongodb.com/docs/mongoid/current/reference/associations/#custom-primary-foreign-keys